code/modules/projectiles/ammo_datums.dm

```diff
@@ -1931,6 +1931,41 @@ GLOBAL_LIST_INIT(no_sticky_resin, typecacheof(list(/obj/item/clothing/mask/faceh
 /datum/ammo/xeno/boiler_gas/corrosive/set_smoke()
 	smoke_system = new /datum/effect_system/smoke_spread/xeno/acid()
 
+/datum/ammo/xeno/larvae_jelly
+	name = "blob of larvae jelly"
+	icon_state = "larvajelly"
+	ping = "ping_x"
+	flags_ammo_behavior = AMMO_XENO|AMMO_SKIPS_ALIENS|AMMO_EXPLOSIVE
+	var/danger_message = "<span class='danger'>A blob of larvae jelly lands with a splat!</span>"
+	armor_type = "bio"
+	accuracy_var_high = 10
+	max_range = 10
+	damage = 1
+	damage_type = STAMINA
+	penetration = 40
+	bullet_color = LIGHT_COLOR_PURPLE
+	reagent_transfer_amount = 5
+
+///Set up the list of reagents the spit transfers upon impact
+/datum/ammo/xeno/larvae_jelly/proc/set_reagents()
```

Comment:
The fact that this proc is redefined three separate times rather than just being on `/datum/ammo/xeno` is... awful.
(Not your fault, just wanted to mention it.)